src/Symfony/Bridge/Twig/Mime/NotificationEmail.php

```diff
@@ -0,0 +1,226 @@
+    public const IMPORTANCE_MEDIUM = 'medium';
+    public const IMPORTANCE_LOW = 'low';
+
+    private const DEFAULT_THEME = '@email/zurb_2/notification';
+
+    private static $defaultTheme = self::DEFAULT_THEME;
+
+    private $theme;
+    private $context = [
+        'importance' => self::IMPORTANCE_LOW,
+        'content' => '',
+        'exception' => false,
+        'action_text' => null,
+        'action_url' => null,
+        'markdown' => false,
+        'raw' => false,
+    ];
+
+    public function __construct(Headers $headers = null, AbstractPart $body = null)
+    {
+        if (!class_exists(CssInlinerExtension::class)) {
+            throw new \LogicException(sprintf('You cannot use "%s" if the CSS Inliner Twig extension is not available; try running "composer require twig/cssinliner-extra".', static::class));
+        }
+
+        if (!class_exists(InkyExtension::class)) {
+            throw new \LogicException(sprintf('You cannot use "%s" if the Inky Twig extension is not available; try running "composer require twig/inky-extra".', static::class));
+        }
+
+        parent::__construct($headers, $body);
+    }
+
+    /**
+     * @return $this
+     */
+    public function markdown()
+    {
+        if (!class_exists(MarkdownExtension::class)) {
+            throw new \LogicException(sprintf('You cannot use "%s" if the Markdown Twig extension is not available; try running "composer require twig/markdown-extra".', __METHOD__));
+        }
+
+        $this->context['markdown'] = true;
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function content(string $content, bool $raw = false)
+    {
+        $this->context['content'] = $content;
+        $this->context['raw'] = $raw;
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function action(string $text, string $url)
+    {
+        $this->context['action_text'] = $text;
+        $this->context['action_url'] = $url;
+
+        return $this;
+    }
+
+    /**
+     * @return self
+     */
+    public function importance(string $importance)
+    {
+        $this->context['importance'] = $importance;
+
+        return $this;
+    }
+
+    /**
+     * @param \Throwable|FlattenException
+     *
+     * @return $this
+     */
+    public function exception($exception)
+    {
+        $exceptionAsString = $this->getExceptionAsString($exception);
+
+        $this->context['exception'] = true;
+        $this->attach($exceptionAsString, 'exception.txt', 'text/plain');
+        $this->importance(self::IMPORTANCE_URGENT);
+
+        if (!$this->getSubject()) {
+            $this->subject($exception->getMessage());
+        }
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function theme(string $themePath = null)
+    {
+        $this->theme = $themePath;
+
+        return $this;
+    }
+
+    public static function setDefaultTheme(string $themePath = null)
```

Comment:
This one is confusing, it sets A theme, but the default theme as default if the value is null? Why not setDefaultTheme without param and a setTheme() with a value and not nullable plus a validation and no fallback to the default theme which is implicit?

Reply:
It sets the default theme globally. Passing `null` means that you want to reset the theme to its default value (I can remove that if it's too confusing). The `setTheme()` method is about changing the theme for that particular instance.

Reply:
To me calling “setDefaultTheme” should set the default theme, so no parameter needed. 👍🏻👌🏻

Reply:
Fixed as we don't have a default theme anymore to set and `theme()` takes a string as an arg.